src/ansys/geometry/core/plotting/plotter.py

```diff
@@ -344,8 +362,44 @@ def add_sketch_polydata(self, polydata_entries: List[pv.PolyData], **plotting_op
             :meth:`Plotter.add_mesh <pyvista.Plotter.add_mesh>` method.
         """
         # Use the default PyAnsys Geometry add_mesh arguments
+        mb = pv.MultiBlock()
         for polydata in polydata_entries:
-            self.scene.add_mesh(polydata, color=EDGE_COLOR, **plotting_options)
+            mb.append(polydata)
+
+        if "clip" in plotting_options:
+            plotting_options.pop("clip")
+            dataset = self.clip(dataset, **plotting_options)
+            plotting_options.pop("normal", None)
+            plotting_options.pop("origin", None)
+
+        self.scene.add_mesh(mb, color=EDGE_COLOR, **plotting_options)
+
+    def clip(
+        self,
+        mesh: Union[pv.PolyData, pv.MultiBlock],
+        normal: str = "x",
+        origin: tuple = None,
+        **plotting_options,
```

Comment:
Why not pass a plane rather than all these options? It's more intuitive

Reply:
Not sure what you mean here, these are the parameters needed to pass to PyVista

Reply:
All those parameters, you can extract them from our Plane object. It makes much more sense to use a "Plane" rather than the PyVista expected inputs. You are missing units there for example...